src/server/system_services/replication_store.js

```diff
@@ -135,16 +135,19 @@ class ReplicationStore {
         return reduced_replications;
     }
 
-    async find_aws_log_based_replication_rules() {
-        dbg.log1('find_aws_log_based_replication_rules: ');
+    async find_log_based_replication_rules() {
+        dbg.log1('find_log_based_replication_rules: ');
         const replications = await this._replicationconfigs.find({ deleted: null });
-
         const aws_reduced_replications = _.filter(replications, repl => repl.log_replication_info?.aws_log_replication_info);
+        const azure_reduced_replications = _.filter(replications, repl => repl.log_replication_info?.azure_log_replication_info);
```

Comment:
you can iterate on the replications array once instead of twice to reduce complexity time